HIP/hip-1195.md

```diff
@@ -0,0 +1,834 @@
+APIs by moving everything into a smart contract.
+
+That is, by avoiding protocol-level changes for every important customization, hooks can greatly streamline innovation
+on a Hiero network while maintaining the performance and integrity of the native services.
+
+## Specification
+
+First we specify how a Hiero network will charge, throttle, and execute EVM hooks. (Non-EVM hook programming models 
+would need their own specifications.)
+
+The protobuf API for hooks in general, and EVM hooks in particular, follows in later sections.
+
+### Charging
+
+A primary concern for EVM hooks is deciding what account pays for the EVM gas upfront before executing the hook. We
+propose two charging patterns that should accommodate most use cases. The **installer** of an EVM hook chooses one of
+the patterns at the time they install it.
+1. `CALLER_PAYS` - The payer of the transaction that triggers the hook pays for its upfront gas cost. They receive the
+normal refund for unused gas.
+2. `CALLER_PAYS_ON_REVERT` - The payer of the triggering transaction again pays for the upfront gas cost, but receives
+a _full refund_ if the hook succeeds (in sense appropriate to the extension point). On success, a designated account
+that signed the lambda's installation pays for the gas actually consumed.
+
+Regardless of the charging pattern, a triggering transaction can always set an explicit gas limit for an EVM hook's
+execution, just as with a contract call. If no explicit limit is set on the transaction, the networks checks if the
+EVM was installed with a default gas limit. If there is no default limit for the hook, the network uses a global
+default, for example `hooks.evm.defaultGasLimit=25_000`.
+
+We propose the same gas price for EVM hook execution as for other contract operations. However, unlike contract calls,
+which are charged purely in gas, hook executions are already "gated" by the fee of their triggering transaction.
+So it makes sense to waive the intrinsic gas cost of their execution. We propose adding two more properties,
+
+```
+hooks.evm.pureIntrinsicGasCost=0
+hooks.evm.lambdaIntrinsicGasCost=0
+```
+
+### Throttling
+
+We propose EVM hooks be subject to the same gas throttle as top-level contract calls. Specifically, when an EVM hook
+executes, its initial EVM sender address is the payer of the referencing transaction. If this payer is throttle exempt,
+no throttles are applied. Otherwise, if the network is at capacity for gas usage, EVM hook execution will be throttled
+on that basis and the triggering transaction fail with a status of `EVM_HOOK_EXECUTION_THROTTLED`.
+
+### EVM hook execution
+
+The first message frame for the EVM transaction that executes an EVM hook will be a logical `DELEGATECALL` from the
+**hook system contract address `0x16d`** to the hook's implementing `ContractID`, sent by the payer of the upfront gas
+cost for the hook's execution.
```

Comment:
> sent by the payer of the upfront gas cost for the hook's execution.

I think _payer_ is misleading here because in some cases it can only be defined later after the transaction is executed (i.e. if CALLER_PAYS_ON_REVERT is enabled).

To make the privileges model clearer, I suggest executing all calls as if they were executed by separate entities (the hook system contract, the hook, the transaction sender, etc.) rather than assuming the privileges of the system contract or the hook installer account.

Reply:
Removed the `CALLER_PAYS[_ON_REVERT]` per TSC discussion.